api/src/main/java/uk/gov/dhsc/htbhf/claimant/message/MessageProcessor.java

```diff
@@ -75,18 +78,33 @@ private MessageStatus processMessage(Message message, MessageTypeProcessor messa
             MessageStatus status = messageTypeProcessor.processMessage(message);
             if (status == MessageStatus.COMPLETED) {
                 messageRepository.delete(message);
+            } else {
+                updateMessageWithStatusAndIncrementCount(message, status);
             }
-            // TODO: HTBHF-1285: handle other message statuses
+
             return status;
         } catch (RuntimeException e) {
             // TODO: HTBHF-1285: expose ErrorHandler.constructExceptionDetail
             //                  from java common and include it in error handling
             log.error("Unable to process message with id {}", message.getId(), e);
-            // TODO: HTBHF-1285: increment delivery count and message timestamp
-            return MessageStatus.ERROR;
+            MessageStatus errorStatus = ERROR;
```

Comment:
We can replace these two lines with updateMessagesToErrorAndIncrementCount

Reply:
That method can validly take different statuses so I'll leave it like that. I've also added ERROR as a variable as this is what's returned from this method.